### ISSUE TITLE
[MINOR] OrbitalCameraSystem leaks window listeners when attach() is called multiple times

### ISSUE THREAD

**Issue by ormidales** [ormidales/microgl#330]
**Category**: security
**Severity**: minor
**File(s)**: `src/core/ecs/systems/OrbitalCameraSystem.ts` (attach, detach)

#### Description
`attach()` internally calls `detach()` to clean up previous canvas-level listeners, but `detach()` only removes listeners from the stored `this.canvas` reference. The `mouseup` and `touchend` handlers registered on `window` are never removed when `attach()` is called a second time with a new canvas, because `this.canvas` is overwritten before `detach()` has a chance to run the window cleanup. This causes window-level handlers to accumulate across scene reloads, producing ghost inputs and a growing memory leak.

#### Problematic code example
```ts
attach(canvas: HTMLCanvasElement): void {
  this.detach();        // removes old canvas listeners only
  this.canvas = canvas; // this.canvas now points to the new canvas —
                        // old window mouseup/touchend handlers are orphaned
  // ...
  window.addEventListener('mouseup', this.onMouseUp);
  window.addEventListener('touchend', this.onTouchEnd, ...);
}
```

#### Suggested fix
Track window listener registration with a boolean flag and always clean up in `detach()` regardless of canvas state.

```ts
private windowListenersAttached = false;

detach(): void {
  if (this.canvas) {
    this.canvas.removeEventListener('mousedown', this.onMouseDown);
    this.canvas.removeEventListener('mousemove', this.onMouseMove);
    this.canvas.removeEventListener('touchstart', this.onTouchStart, OrbitalCameraSystem.PASSIVE_EVENT_OPTIONS);
    this.canvas.removeEventListener('touchmove', this.onTouchMove, OrbitalCameraSystem.NON_PASSIVE_EVENT_OPTIONS);
    this.canvas.removeEventListener('wheel', this.onWheel, OrbitalCameraSystem.NON_PASSIVE_EVENT_OPTIONS);
    this.canvas = null;
  }
  if (this.windowListenersAttached) {
    window.removeEventListener('mouseup', this.onMouseUp);
    window.removeEventListener('touchend', this.onTouchEnd, OrbitalCameraSystem.PASSIVE_EVENT_OPTIONS);
    this.windowListenersAttached = false;
  }
}
```

#### Acceptance criteria
- [ ] `detach()` removes `mouseup` and `touchend` from `window` unconditionally when they were registered
- [ ] Multiple sequential `attach()` calls leave exactly one set of window-level handlers
- [ ] `window` access is guarded with `typeof window !== 'undefined'` for SSR / Worker compatibility
- [ ] Tests assert that re-attaching then detaching leaves zero window-level handlers
